src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java

```diff
@@ -303,6 +304,23 @@ public NativeMemoryAllocation get(NativeMemoryEntryContext<?> nativeMemoryEntryC
         return cache.get(nativeMemoryEntryContext.getKey(), nativeMemoryEntryContext::load);
     }
 
+    /**
+     * Returns the NativeMemoryAllocation associated with given index
+     * @param indexName name of OpenSearch index
+     * @return NativeMemoryAllocation associated with given index
+     */
+    public Optional<NativeMemoryAllocation> getIndexMemoryAllocation(String indexName) {
```

Comment:
nit: I think our convention is to have params in public methods as final. not a critical though, just for future PRs